### ISSUE TITLE
event strings are should be enums

### ISSUE THREAD

**Issue by flavono123** [kubernetes-sigs/karpenter#1897]
### Description

**What problem are you trying to solve?**

to get all reason from karpenter controllers', for now, have to parse the codes since they are as string.
these should be constants i guess (and i could if some naming conventions are)

**How important is this feature to you?**

not urgent but required for monitoring events by controllers

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by jonathan-innis** [comment]
Makes sense to me -- we should probably just gather the reasons in a shared package like in the apis directory so that they have a very small dependency chain and they can be easily accessed everywhere without pulling in too many other dependencies

**Comment by flavono123** [comment]
ok.
can i try a PR for this? i'm asking if i can, do i have to label somewhat to this issue?

**Comment by flavono123** [comment]
have learned from other issues

/assign
